wkcuber/wkcuber/tile_cubing.py

```diff
@@ -292,39 +292,81 @@ def tile_cubing(
         return
 
     # Determine tile size from first matching file
-    tile_size = image_reader.read_dimensions(arbitrary_file)
+    num_x, num_y = image_reader.read_dimensions(arbitrary_file)
+    num_z = max_dimensions["z"] - min_dimensions["z"] + 1
     num_channels = image_reader.read_channel_count(arbitrary_file)
     logging.info(
         "Found source files: count={} with tile_size={}x{}".format(
-            file_count, tile_size[0], tile_size[1]
+            file_count, num_x, num_y
         )
     )
     if args is None or not hasattr(args, "dtype") or args.dtype is None:
         dtype = image_reader.read_dtype(arbitrary_file)
+    else:
+        dtype = args.dtype
+
+    target_ds = Dataset.get_or_create(target_path, scale=(1, 1, 1))
+    is_segmentation_layer = layer_name == "segmentation"
+    if is_segmentation_layer:
+        target_layer = target_ds.get_or_add_layer(
+            layer_name,
+            LayerCategories.SEGMENTATION_TYPE,
+            dtype_per_channel=dtype,
+            num_channels=num_channels,
+            largest_segment_id=0,
```

Comment:
Updating the `largest_segment_id` in the `write` function does not work because `View.write()` does not know the dataset. However, for this particular case, we can simply compute the `largest_segment_id` for each buffer in `tile_cubing_job` and then update the id once all jobs have finished cubing. (We do the same thing in `cubing` since recently)